### PR TITLE
fix(badge): import `UserNotifications` to support MacOS

### PR DESCRIPTION
**Pull request by kainosnoema**
## Description

We're maintaining a private fork of Capacitor with slight changes for Mac OS compatibility with WKWebView (works great, actually), and the Badge plugin works well with one small tweak: need to import `UserNotifications` explicitly since it's not imported by default by Mac OS (as it is in iOS). I realize you're not attempting to support Mac OS here but it's a small, harmless addition that would be great to get in—otherwise we'll have to maintain our own fork of this repo.

## Pull request checklist

- [ ] The changes have been tested successfully.
- [x] A changeset has been created (`npm run changeset`).

### CODE REVIEW

**Comment by kainosnoema** [comment]
I'm having trouble getting the verification to run, getting some errors related to Java versions (have 1.8.0_361 installed). Maybe someone can help point me in the right direction of resolving it:

```
@capawesome/capacitor-badge:verify:android: FAILURE: Build failed with an exception.
@capawesome/capacitor-badge:verify:android:
@capawesome/capacitor-badge:verify:android: * What went wrong:
@capawesome/capacitor-badge:verify:android: A problem occurred configuring root project 'android'.
@capawesome/capacitor-badge:verify:android: > Could not resolve all files for configuration ':classpath'.
@capawesome/capacitor-badge:verify:android:    > Could not resolve com.android.tools.build:gradle:7.4.1.
@capawesome/capacitor-badge:verify:android:      Required by:
@capawesome/capacitor-badge:verify:android:          project :
@capawesome/capacitor-badge:verify:android:       > No matching variant of com.android.tools.build:gradle:7.4.1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.5' but:
@capawesome/capacitor-badge:verify:android:           - Variant 'apiElements' capability com.android.tools.build:gradle:7.4.1 declares a library, packaged as a jar, and its dependencies declared externally:
@capawesome/capacitor-badge:verify:android:               - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
@capawesome/capacitor-badge:verify:android:               - Other compatible attribute:
@capawesome/capacitor-badge:verify:android:                   - Doesn't say anything about org.gradle.plugin.api-version (required '7.5')
```

**Comment by robingenz** [comment]
> I'm having trouble getting the verification to run, getting some errors related to Java versions (have 1.8.0_361 installed). Maybe someone can help point me in the right direction of resolving it:
> 
> ```
> @capawesome/capacitor-badge:verify:android: FAILURE: Build failed with an exception.
> @capawesome/capacitor-badge:verify:android:
> @capawesome/capacitor-badge:verify:android: * What went wrong:
> @capawesome/capacitor-badge:verify:android: A problem occurred configuring root project 'android'.
> @capawesome/capacitor-badge:verify:android: > Could not resolve all files for configuration ':classpath'.
> @capawesome/capacitor-badge:verify:android:    > Could not resolve com.android.tools.build:gradle:7.4.1.
> @capawesome/capacitor-badge:verify:android:      Required by:
> @capawesome/capacitor-badge:verify:android:          project :
> @capawesome/capacitor-badge:verify:android:       > No matching variant of com.android.tools.build:gradle:7.4.1 was found. The consumer was configured to find a runtime of a library compatible with Java 8, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.gradle.plugin.api-version' with value '7.5' but:
> @capawesome/capacitor-badge:verify:android:           - Variant 'apiElements' capability com.android.tools.build:gradle:7.4.1 declares a library, packaged as a jar, and its dependencies declared externally:
> @capawesome/capacitor-badge:verify:android:               - Incompatible because this component declares an API of a component compatible with Java 11 and the consumer needed a runtime of a component compatible with Java 8
> @capawesome/capacitor-badge:verify:android:               - Other compatible attribute:
> @capawesome/capacitor-badge:verify:android:                   - Doesn't say anything about org.gradle.plugin.api-version (required '7.5')
> ...
> ```

You need to use Java 11. Let’s the if the pipeline run is successful.

**Comment by robingenz** [comment]
@kainosnoema Please note that all plugins are already updated to Capacitor 5 for the next release. just FYI

File: .changeset/tender-kids-notice.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@capawesome/capacitor-badge": patch
+---
+
+fix(badge): add UserNotifications import to fix build on Mac OS
```
Comment (robingenz) [inline .changeset/tender-kids-notice.md]:
```suggestion
fix(badge): import `UserNotifications` to support MacOS
```

**Pull request merged**